src/components/tables/Table.tsx

```diff
@@ -168,9 +168,17 @@ export class Table extends React.Component<ITableProps> {
         const {tableCompositeState} = this.props;
 
         if (this.hasTableCompositeStateChanged(tableCompositeState, nextProps.tableCompositeState)) {
-            // if the change occurs outside the navigation (per page, pagination), reset the pagination to 0
-            const shouldResetPage = tableCompositeState.page === nextProps.tableCompositeState.page
-                && tableCompositeState.perPage === nextProps.tableCompositeState.perPage;
+            const shouldResetPage =

```

Comment:
What does this do exactly / how does it work? I feel like it would be annoying to reset paging for every case where we add an entry. Maybe have some prop to toggle it would be nice.

Reply:
I think you're right the logic needs to be reworked and more flexible, you can try the example and give me some feedback if you'd like